implementations/js-hyperjump/bowtie_hyperjump.js

```diff
@@ -24,42 +18,138 @@ function send(data) {
-const keywordsNotInSchemaSkippedTests = [
-  "naive replacement of $ref with its destination is not correct",
-  "$ref prevents a sibling id from changing the base uri",
-  "$ref prevents a sibling $id from changing the base uri",
-].reduce((acc, description) => {
-  acc[description] = keywordsNotInSchemaMessage;
-  return acc;
-}, {});
-
-const boundaryCrossingMessage =
-  "JSON pointers that cross schema resource boundaries are not suppported. There might be a way to solve this, but because this functionality has been removed from the spec and there is no good reason to do this in any version of the spec, it will probably never be fixed.";
-const boundaryCrossingSkippedTests = {
-  "base URI change - change folder in subschema": boundaryCrossingMessage,
-};
+async function setup() {
+  // skip
+  const keywordsNotInSchemaMessage =
+    "Ignoring schema meta-data keywords in places that are not schemas (such as a $id in a const) is not supported. Because this implementation is dialect agnostic, there's no way to know whether a location is a schema or not. Especially because there's no reason for a schema to use keywords in places that aren't schemas, I'm not concerned about making it work.";
 
-const fileSchemeMessage =
-  "Self-identifying with a `file:` URI is not allowed for security reasons.";
-const fileSchemeSkippedTests = {
-  "id with file URI still resolves pointers - *nix": fileSchemeMessage,
-  "id with file URI still resolves pointers - windows": fileSchemeMessage,
-  "$id with file URI still resolves pointers - *nix": fileSchemeMessage,
-  "$id with file URI still resolves pointers - windows": fileSchemeMessage,
-};
+  const boundaryCrossingMessage =
+    "JSON pointers that cross schema resource boundaries are not suppported. There might be a way to solve this, but because this functionality has been removed from the spec and there is no good reason to do this in any version of the spec, it will probably never be fixed.";
+  const boundaryCrossingSkippedTests = {
+    "base URI change - change folder in subschema": boundaryCrossingMessage,
+  };
 
-const modernSkippedTests = { ...fileSchemeSkippedTests };
-const legacySkippedTests = {
-  ...boundaryCrossingSkippedTests,
-  ...keywordsNotInSchemaSkippedTests,
-  ...fileSchemeSkippedTests,
-};
+  if (hyperjump_version >= "1.7.0") {
+    await Promise.all([
+      import("@hyperjump/json-schema/draft-04"),
+      import("@hyperjump/json-schema/draft-06"),
+      import("@hyperjump/json-schema/draft-07"),
+      import("@hyperjump/json-schema/draft-2019-09"),
+    ]);
+    const module = await import("@hyperjump/json-schema/draft-2020-12");
```

Comment:
These imports are needed for all v1, not just >= 1.7.0.

The only difference from 1.7.0 forward is that `addSchema` was replaced with `registerSchema`. Actually, `addSchema` still works, it's just deprecated.